packages/material-ui-lab/src/TreeView/TreeView.test.js

```diff
@@ -174,31 +218,32 @@ describe('<TreeView />', () => {
       const [, setState] = React.useState(1);
 
       return (
-        <TreeView>
-          <TreeItem
-            nodeId="one"
-            label="one"
-            data-testid="one"
-            onFocus={() => {
-              setState(Math.random);
-            }}
-          >
+        <TreeView
+          onFocus={() => {
+            setState(Math.random);
+          }}
+          id="tree"
+        >
+          <TreeItem nodeId="one" label="one" data-testid="one">
             <TreeItem nodeId="two" label="two" data-testid="two" />
           </TreeItem>
         </TreeView>
       );
     }
 
-    const { getByText, getByTestId } = render(<MyComponent />);
+    const { getByText, getByRole } = render(<MyComponent />);
 
     fireEvent.click(getByText('one'));
-    expect(getByTestId('one')).toHaveFocus();
-    fireEvent.keyDown(getByTestId('one'), { key: 'ArrowDown' });
-    expect(getByTestId('two')).toHaveFocus();
-    fireEvent.keyDown(getByTestId('two'), { key: 'ArrowUp' });
-    expect(getByTestId('one')).toHaveFocus();
-    fireEvent.keyDown(getByTestId('one'), { key: 'ArrowDown' });
-    expect(getByTestId('two')).toHaveFocus();
+    // Clicks would normally focus tree
+    getByRole('tree').focus();
+
+    expect(getByRole('tree')).to.have.attribute('aria-activedescendant', 'tree-one');
+    fireEvent.keyDown(getByRole('tree'), { key: 'ArrowDown' });
+    expect(getByRole('tree')).to.have.attribute('aria-activedescendant', 'tree-two');
```

Comment:
We should verify that the tree is still focused. 

Might make sense to add a `toHaveVirtualFocus`  (better name required). That checks if the asserted element is marked as `aria-activedescendant` from the `document.activeElement`.

Reply:
Sounds good. I can't think of a name right now so it'll be called toHaveVirtualFocus ;) but will rename before merge.